front-end/app/service/basket.service.ts

```diff
@@ -1,6 +1,54 @@
 import { Injectable } from '@angular/core';
 import { BasketService as SharedBasketService } from '../../../shared/services/basket-service';
 
+const STORAGE_KEY = 'basket';
+
 @Injectable()
 export class BasketService extends SharedBasketService {
+    constructor() {
+        super();
+        this.loadItems();
+    }
+
+    addToBasket(item: BasketItem): void {
+        super.addToBasket(item);
+        this.saveItems();
+    }
+
+    increase(item: OrderLineItem) {
+        super.increase(item);
+        this.saveItems();
+    }
+
+    decrease(item: OrderLineItem) {
+        super.decrease(item);
+        this.saveItems();
+    }
+
+    removeItem(item: OrderLineItem): void {
+        super.removeItem(item);
+        this.saveItems();
+    }
+
+    removeAll(): void {
+        super.removeAll();
+        this.saveItems();
+    }
+
+    private loadItems(): void {
+        let basket = JSON.parse(localStorage.getItem(STORAGE_KEY) || null);
+        const HOUR = 60 * 60 * 1000;
+        if (basket) {
```

Comment:
In our scenario, basket will either be an object or it will be null. I generally prefer to be as explicit as possible and write basket !== null because this avoids any coercion.